src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -91,7 +92,17 @@ public async Task<GeneratedCodeWorkspace> ExecuteAsync(InputNamespace rootNamesp
             Directory.CreateDirectory(Configuration.OutputFolder);
             var project = await GeneratedCodeWorkspace.Create(Configuration.AbsoluteProjectFolder, Configuration.OutputFolder, Configuration.SharedSourceFolders);
             var sourceInputModel = new SourceInputModel(await project.GetCompilationAsync(), await ProtocolCompilationInput.TryCreate());
-            await LowLevelTarget.ExecuteAsync(project, rootNamespace, sourceInputModel, true);
+
+            if (Configuration.AzureArm)
+            {
+                InputTypeTransformer.Transform(rootNamespace);
```

Comment:
please add a TODO item here because in the future we should never need to transform anything
Everything we need to change should happen on the tsp instead of being done in the generator.

Reply:
I am not 100% sure if we never need to transform anything.
This is a workaround to handle internal operations for now.
There might be cases in CodeModelTransformer, we need to handle for TSP as well, just we have not met yet.

Reply:
never mind, moved the logic to `MgmtOutputLibrary`, and we can add `InputTypeTransformer` until we need it.

Reply:
Ok. We still need the transformer since we are manipulating the operations in client.

Reply:
Added TODO to remove this once we have a better way to remove operation, and we can discuss later whether we need the transformer.